### PR TITLE
[dev2] Removed `**:download_cache` configurations and `CachedFileDownloader` class

### LINKED ISSUE #11515: [develop2] [question] Reconsider ``CachedFileDownloader``?
Maybe ask for feedback to the Tribe? I am not fully convinced about the value, maybe retire it in 2.0 while we investigate concurrency in the cache?

### PR DESCRIPTION
**Pull request by franramirez688**
Changelog: Feature: Removed `CachedFileDownloader` class, and `core.download:download_cache` and `tools.files.download:download_cache` configurations.
Closes: https://github.com/conan-io/conan/pull/11650
Docs: omit

### CODE REVIEW

File: conans/client/downloaders/download.py [hunk 1]
```diff
@@ -1,11 +1,9 @@
-from conans.client.downloaders.cached_file_downloader import CachedFileDownloader
 from conans.client.downloaders.file_downloader import FileDownloader
 
 
-def run_downloader(requester, verify, retry, retry_wait, download_cache,
-                   **kwargs):
+def run_downloader(requester, verify, retry, retry_wait, download_cache, **kwargs):
```
Comment (franramirez688) [inline conans/client/downloaders/download.py:4]: Keeping the wrapper, but do we want to keep `download_cache` as well?

Comment (memsharded) [reply conans/client/downloaders/download.py:4]: Yes, you can introduce an assert to check it is never True

File: conans/test/integration/cache/download_cache_test.py [hunk 2]
```diff
@@ -73,6 +73,7 @@ def test_dirty_download(self):
         client.run("install --requires=pkg/0.1@")
         assert "pkg/0.1: Downloaded package" in client.out
 
+    @pytest.mark.skip()  # Skipping test because of tools.files.download:download_cache deletion
```
Comment (memsharded) [inline conans/test/integration/cache/download_cache_test.py]: why not removing the whole file?

Comment (franramirez688) [reply conans/test/integration/cache/download_cache_test.py]: yeah, you're right

**Review by memsharded — APPROVED** [review]

**Review by lasote — APPROVED** [review]

**Pull request merged**